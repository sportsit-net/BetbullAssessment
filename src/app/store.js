import { configureStore } from '@reduxjs/toolkit';
import loginReducer from '../components/Login/loginSlice';
import gameReducer from '../components/Game/gameSlice';
import scoreCardReducer from '../components/ScoreCard/scorecardSlice';

export default configureStore({
  reducer: {
    login: loginReducer,
    game: gameReducer,
    scorecard:scoreCardReducer
  },
});
