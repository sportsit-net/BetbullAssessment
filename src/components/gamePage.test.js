import React from 'react'
import GamePage from './GamePage';
import { shallow,configure,mount} from 'enzyme';
import {Provider} from 'react-redux';
import store from './../store'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });
 describe('<GamePage /> with no props',()=>{
    const wrapper = mount(<Provider store={store}><GamePage /></Provider>);
   const gamePage= wrapper.find("[data-test='gamePageTest']")
    it('game page card count',()=>{
  const checkCard=  gamePage.hasClass("SMG-card");
  expect(checkCard).toBe(true)
    })
 })