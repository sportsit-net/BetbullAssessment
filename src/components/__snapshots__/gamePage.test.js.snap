// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamePage /> with no props should match the snapshot 1`] = `"<div class=\\"SM-gamePage\\"><div role=\\"alert\\" class=\\"fade alert alert-info show\\">You have 3 chances left</div><div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"3\\"/><div class=\\"SM-front s3\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"9\\"/><div class=\\"SM-front s9\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"6\\"/><div class=\\"SM-front s6\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"7\\"/><div class=\\"SM-front s7\\"></div><div class=\\"SM-back\\"></div></div></div><div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"2\\"/><div class=\\"SM-front s2\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"5\\"/><div class=\\"SM-front s5\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"7\\"/><div class=\\"SM-front s7\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"8\\"/><div class=\\"SM-front s8\\"></div><div class=\\"SM-back\\"></div></div></div><div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"9\\"/><div class=\\"SM-front s9\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"2\\"/><div class=\\"SM-front s2\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"3\\"/><div class=\\"SM-front s3\\"></div><div class=\\"SM-back\\"></div></div><div class=\\"SM-card\\"><input type=\\"checkbox\\" value=\\"8\\"/><div class=\\"SM-front s8\\"></div><div class=\\"SM-back\\"></div></div></div></div>"`;
